Update Subaru Sensors

G-Two opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
* Change "EV Time to Fully Charged" type to datetime object (HA 2022.2)
* Handle invalid sensors values sometimes returned by Subaru API
* Depends on #66403

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->
- This PR is related to issue:  https://github.com/home-assistant/core/pull/48193#discussion_r602869116

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [x] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

G-Two committed:
Update sensor.py

* Change "EV Time to Fully Charged" type to datetime object (HA 2022.2)

* Validate types before accessing dict entries

* Test handling of invalid data from Subaru

* Bump to subarulink 0.4.2

epenet commented on homeassistant/components/subaru/sensor.py:
```diff
@@ -270,16 +269,22 @@ def available(self):
-            else:
-                value = int(value)
-        return value
+        if isinstance(self.coordinator.data, dict):
+            value = self.coordinator.data.get(self.vin)[VEHICLE_STATUS].get(
+                self.data_field
+            )
```
Would this style work?
```suggestion
        if isinstance(data := self.coordinator.data, dict):
            value = data.get(self.vin)[VEHICLE_STATUS].get(self.data_field)
```

G-Two replied:
yes, thanks!

Bre77 committed:
Bump Advantage Air to 0.2.6 (#65849)

davet2001 committed:
Fix missing refactors of EntityCategory.XXX (#66379)

* Fix missing refactors of EntityCategory.XXX

* Fix entity_category refactor for homewizard

G-Two committed:
Incorporate style suggestion

chemelli74 committed:
Fix mesh role for Fritz old devices (#66369)

allenporter committed:
Reset the stream backoff timeout when the url updates (#66426)

Reset the stream backoff timeout when the url updates, meant to improve the retry behavior
for nest cameras. The problem is the nest url updates faster than the stream reset time
so the wait timeout never resets if there is a temporarily problem with the new url. In particular
this *may* help with the flaky cloud nest urls, but seems more correct otherwise.

raman325 committed:
revert change in vizio logic to fix bug (#66424)

epenet committed:
Fix access to hass.data in hdmi-cec (#66504)

Co-authored-by: epenet <epenet@users.noreply.github.com>

Bre77 committed:
Bump Advantage Air 0.3.0 (#66488)

thecode committed:
Increase switcher_kis timeouts (#66465)

uSlackr committed:
Correct modbus address limits (#66367)

bdraco committed:
Fix flux_led turn on with slow responding devices (#66527)

epenet committed:
Revert "Fix raspihats callbacks (#64122)" (#66517)

Co-authored-by: epenet <epenet@users.noreply.github.com>

dgomes committed:
Fix utility meter restore state (#66490)

* Address #63874

* avoid setting _last_period to None

* name is always set in discovery

* ValueError never happens only DecimalException

* async_tariff_change tracks state change - state machine will not pass a None

* test we only reset one utility_meter

* test corrupted restored state

* pretty sure _current_tariff doesn't change from init until here

* missing assert

* Revert "async_tariff_change tracks state change - state machine will not pass a None"

This reverts commit 24fc04a964139e5cfecbfa20f91e2d30ab145d77.

* address review comment

* always a Decimal

epenet committed:
Add tests for samsungtv diagnostics (#66563)

* Add tests for samsungtv diagnostics

* Adjust coveragerc

* Adjust type hints

Co-authored-by: epenet <epenet@users.noreply.github.com>

frenck committed:
Fix Tuya Covers without state in their control data point (#66564)

iMicknl committed:
Always create a new session in ConfigFlow in Overkiz integration (#66602)

elupus committed:
Add fallback for serialnumber (#66553)

elupus committed:
Add fallback for serialnumber (#66553)

allenporter committed:
Override and disable nest stream `unavailable` behavior (#66571)

allenporter committed:
Override and disable nest stream `unavailable` behavior (#66571)

bdraco committed:
Deduplicate flux_led title and CONF_NAME (#66598)

bdraco committed:
Enable dhcp flows for wiz registered devices (#66595)

bdraco committed:
Enable dhcp flows for emonitor registered devices (#66584)

bdraco committed:
Switch steamist to use integration discovery (#66578)

bdraco committed:
Switch elkm1 to use integration discovery (#66572)

frenck committed:
Update plugwise 0.16.4 (#66613)

bdraco committed:
Switch senseme to use integration discovery (#66576)

Commit:
[ci skip] Translation update

bdraco committed:
Switch flux_led to use integration discovery (#66574)

bdraco committed:
Enable dhcp flows for steamist registered devices (#66593)

bdraco committed:
Enable dhcp flows for hunterdouglas_powerview registered devices (#66587)

bdraco committed:
Enable dhcp flows for senseme registered devices (#66590)

bdraco committed:
Enable dhcp flows for screenlogic registered devices (#66591)

bdraco committed:
Enable dhcp flows for isy994 registered devices (#66588)

marcelveldt committed:
Bump aiohue to version 4.1.2 (#66609)

marcelveldt committed:
Bump aiohue to version 4.1.2 (#66609)

balloob committed:
Bumped version to 2022.2.7

jjlawren committed:
Backport #66399 to rc (#66625)

balloob committed:
Merge pull request #66624 from home-assistant/rc

raman325 committed:
Fix zwave_js device condition bug (#66626)

epenet committed:
Cleanup samsungtv tests (#66570)

* Drop unused init method

* Add type hints to media_player tests

* Adjust test_init

* Adjust media_player

* Add type hints to conftest

* Use Mock in test_media_player

* Use lowercase in test_init

* Use relative import in diagnostics

* Add type hints to config_flow

* Adjust coveragerc

* Make gethostbyname autouse

* Cleanup gethostbyname and remote fixtures

* Drop unused fixtures

* Undo type hints and usefixtures on media_player

* Undo type hints and usefixtures in test_init

* Undo type hints in conftest

* Undo usefixtures in test_config_flow

* Format

Co-authored-by: epenet <epenet@users.noreply.github.com>

epenet committed:
Cleanup samsungtv tests (#66570)

* Drop unused init method

* Add type hints to media_player tests

* Adjust test_init

* Adjust media_player

* Add type hints to conftest

* Use Mock in test_media_player

* Use lowercase in test_init

* Use relative import in diagnostics

* Add type hints to config_flow

* Adjust coveragerc

* Make gethostbyname autouse

* Cleanup gethostbyname and remote fixtures

* Drop unused fixtures

* Undo type hints and usefixtures on media_player

* Undo type hints and usefixtures in test_init

* Undo type hints in conftest

* Undo usefixtures in test_config_flow

* Format

Co-authored-by: epenet <epenet@users.noreply.github.com>

bdraco committed:
Enable dhcp flows for goalzero registered devices (#66586)

bdraco committed:
Enable dhcp flows for broadlink registered devices (#66582)

bdraco committed:
Switch tplink to use integration discovery (#66575)

bdraco committed:
Switch ezviz to use integration discovery (#66579)

bdraco committed:
Enable dhcp flows for flux_led registered devices (#66585)

tschnilo committed:
Add vicare sensors (#63339)

* Add sensors

for:
- solar collector 
- power consumption

* Update sensor.py

* Update sensor.py

* Update sensor.py

* Update sensor.py

bdraco committed:
Enable dhcp flows for elkm1 registered devices (#66583)

emontnemery committed:
Include changes in EVENT_DEVICE_REGISTRY_UPDATED (#66641)

bdraco committed:
Enable dhcp flows for samsungtv registered devices (#66589)

milanmeu committed:
Add binary sensor platform to Aseko (#66643)

frenck committed:
Fix Plugwise auto HVAC mode (#66639)

* Fix Plugwise auto hvac mode

* Clean up set HVAC

SaSa1983 committed:
Fix scaling of numeric Tuya values (#66644)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

SaSa1983 committed:
Fix scaling of numeric Tuya values (#66644)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

mfugate1 committed:
Refactor sleepiq as async with config flow (#64850)

Co-authored-by: J. Nick Koston <nick@koston.org>

epenet committed:
Add epenet to samsungtv codeowners (#66654)

Co-authored-by: epenet <epenet@users.noreply.github.com>

epenet committed:
Fix try_connect in samsungtv (#66653)

* Fix try_connect in samsungtv

* Use try...else

* Adjust try...else

* Undo try...else

Co-authored-by: epenet <epenet@users.noreply.github.com>

dgomes committed:
add entity_category (#66377)

cgtobi committed:
Improve handling of cloud hook registration (#65664)

Signed-off-by: cgtobi <cgtobi@gmail.com>

balloob committed:
Add Google local indicator (#66610)

frenck committed:
Add current temp fallback in Tuya climate (#66664)

frenck committed:
Add current temp fallback in Tuya climate (#66664)

bramkragten committed:
Keep TTS media browser params in identifier (#66663)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

Kane610 committed:
Add Button platform to deCONZ integration (#65700)

* Improve scene platform

* Add button platform, tests and fix tests affected by new entities existing

* Remove unnused property

* Bump dependency to v87

bdraco committed:
Fix missing effects on dimmable WiZ bulbs (#66665)

bdraco committed:
Teach _async_abort_entries_match about entry options (#66662)

balloob committed:
Bump aiohue to 4.2.0 (#66670)

balloob committed:
Bump aiohue to 4.2.0 (#66670)

craigjmidwinter committed:
Bump pysher to 1.0.7 (#59445)

* Fix Goalfeed integration

See https://github.com/deepbrook/Pysher/issues/62

* update requirements

* Update pysher, remove websocket-client requirement for goalfeed integration

craigjmidwinter committed:
Bump pysher to 1.0.7 (#59445)

* Fix Goalfeed integration

See https://github.com/deepbrook/Pysher/issues/62

* update requirements

* Update pysher, remove websocket-client requirement for goalfeed integration

balloob committed:
Cloud to avoid setting up Alexa/Google during setup phase (#66676)

balloob committed:
Cloud to avoid setting up Alexa/Google during setup phase (#66676)

frenck committed:
Do not pass client session to Brunt (#66671)

frenck committed:
Do not pass client session to Brunt (#66671)

bieniu committed:
Fix type of value in MQTT binary sensor (#66675)

bieniu committed:
Fix type of value in MQTT binary sensor (#66675)

epenet committed:
Fix SamsungTVWS mocking in samsungtv tests (#66650)

Co-authored-by: epenet <epenet@users.noreply.github.com>

epenet committed:
Fix SamsungTVWS mocking in samsungtv tests (#66650)

Co-authored-by: epenet <epenet@users.noreply.github.com>

bramkragten committed:
Allow metadata in service call data (#66672)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>
Co-authored-by: Erik <erik@montnemery.com>

marcelveldt committed:
Fix last_activated timestamp on Hue scenes (#66679)

marcelveldt committed:
Fix last_activated timestamp on Hue scenes (#66679)

epenet committed:
Fix token refresh in samsungtv (#66533)

epenet committed:
Fix token refresh in samsungtv (#66533)

epenet committed:
Fix side_effect patching in samsungtv tests (#66651)

Co-authored-by: epenet <epenet@users.noreply.github.com>

balloob committed:
Bumped version to 2022.2.8

emontnemery committed:
Remove duplicated options from input_select (#66680)

frenck committed:
Update plugwise 0.16.5 (#66684)

epenet committed:
Fix slow samsungtv test (#66696)

emontnemery committed:
Correct MQTT binary_sensor and sensor state restoring (#66690)

emontnemery committed:
Correct MQTT binary_sensor and sensor state restoring (#66690)

Bre77 committed:
Code Quality improvements for Aussie Broadband (#65408)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

balloob committed:
Merge pull request #66686 from home-assistant/rc

emontnemery committed:
Improve MQTT binary_sensor test (#66688)

Commit:
[ci skip] Translation update

Bre77 committed:
Bump Advantage Air to 0.3.1 (#66699)

rborkow committed:
Bump pyeconet version for Gen 5 Water Heater Support (#66691)

* Bump pyeconet version for Gen 5 Water Heater Support

* Update requirements files

starkillerOG committed:
Netgear add reboot button (#65706)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>
Co-authored-by: Franck Nijhof <git@frenck.dev>

eavanvalkenburg committed:
Brunt package to 1.2.0 (#66722)

raman325 committed:
Add inclusion state to zwave_js/network_status WS API cmd (#65398)

frenck committed:
Update google-cloud-texttospeech to 2.10.0 (#66726)

epenet committed:
Add device info to samsungtv diagnostics (#66728)

* Add device-info to samsungtv diagnostics

* Adjust tests

Co-authored-by: epenet <epenet@users.noreply.github.com>

emontnemery committed:
Don't allow creating or updating input_select with duplicates (#66718)

* Don't allow creating or updating input_select with duplicates

* Simplify error message

* Improve error message

jbouwh committed:
MQTT climate preset_modes rework (#66062)

* MQTT climate preset_modes rework

* Set deprection date to 2022.9 (6 months)

* add valid_preset_mode_configuration for discovery

* Update deprecation date

frenck committed:
Revert "Update google-cloud-texttospeech to 2.10.0" (#66736)

frenck committed:
Use min/max/step from thermostat in Plugwise (#66618)

nbogojevic committed:
Bump withings-api 2.3.2->2.4.0 (#66723)

frenck committed:
Add Python 3.10 to CI (#59729)

frenck committed:
Fix trigger of full CI on dependency bumps (#66738)

CoMPaTech committed:
Plugwise: Update fixtures (#66749)

cdce8p committed:
Remove ThreadPoolExecutor `shutdown` backport (#66735)

ctalkington committed:
Improve roku play media handling (#66429)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

emontnemery committed:
Use pylint disable-next in MQTT (#66758)

thecode committed:
Fix webostv notify service (#66760)

thecode committed:
Fix webostv notify service (#66760)

CoMPaTech committed:
Plugwise update Zigbee addressing fixture data to 64bits (#66761)

bruxy70 committed:
Bump holidays to 0.13 (#66612)

andre-richter committed:
Remove deprecated way of setting fan preset in Vallox (#66655)

felipecrs committed:
Fix Twilio webhook content type (#66561)

epenet committed:
Add type hints in samsungtv tests (#66632)

Co-authored-by: epenet <epenet@users.noreply.github.com>

epenet committed:
Ensure new samsungtv token is updated in the config_entry (#66731)

Co-authored-by: epenet <epenet@users.noreply.github.com>

epenet committed:
Ensure new samsungtv token is updated in the config_entry (#66731)

Co-authored-by: epenet <epenet@users.noreply.github.com>

teharris1 committed:
Bump pyinsteon to 1.0.16 (#66759)

teharris1 committed:
Bump pyinsteon to 1.0.16 (#66759)

epenet committed:
Fix samsung mocks (#66765)

Co-authored-by: epenet <epenet@users.noreply.github.com>

cdce8p committed:
Improve `device_automation` typing (#66621)

emontnemery committed:
Remove use of hass.helpers from MQTT (#66757)

* Remove use of hass.helpers from MQTT

* Tweak

thecode committed:
Handle default notify data in webostv (#66770)

thecode committed:
Handle default notify data in webostv (#66770)

bdraco committed:
Ensure lutron caseta imports set the unique id (#66754)

GitHub-Action committed:
[ci skip] Translation update

bdraco committed:
Fix merge conflict resolution error in flux_led (#66775)

allenporter committed:
Update nest camera tests to use common test fixture (#66192)

cdce8p committed:
Add type ignore error codes [core] (#66773)

cdce8p committed:
Add type ignore error codes [other] (#66781)

allenporter committed:
Bump grpcio to 1.44.0 (#66787)

raman325 committed:
Bump zwave-js-server-python to 0.35.0 (#66785)

* Bump zwave-js-server-python to 0.35.0

* Remove support for new event type which should go in a separate PR

frenck committed:
Add config flow to MJPEG IP Camera (#66607)

davet2001 committed:
Increase helpers.frame test coverage (#65137)

Co-authored-by: Erik Montnemery <erik@montnemery.com>

chemelli74 committed:
Fix wifi switches name for Fritz (#66529)

emontnemery committed:
Add MQTT diagnostics (#66730)

* Add MQTT diagnostics

* Redact device tracker location

* Adjust tests

* Address comments from code review

cdce8p committed:
Add type ignore error codes [A-L] (#66778)

cdce8p committed:
Add type ignore error codes [N-Z] (#66779)

dependabot[bot] committed:
Bump docker/login-action from 1.12.0 to 1.13.0 (#66788)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

rytilahti committed:
Bump python-songpal dependency to 0.14 (#66769)

* Bump python-songpal dependency to 0.14

* Fix tests

* pip_check -1

Co-authored-by: Franck Nijhof <git@frenck.dev>

alengwenus committed:
Bump pypck to 0.7.14 (#66794)

cdce8p committed:
Add type ignore error codes [recorder] (#66780)

rytilahti committed:
Bump python-miio dependency to 0.5.10 (#66782)

ludeeus committed:
Add list to async_delay_save typing (#66795)

cdce8p committed:
Add type ignore error codes [util] (#66777)

cdce8p committed:
Add type ignore error codes [helpers] (#66776)

emontnemery committed:
Correct state restoring for MQTT temperature sensors (#66741)

* Correct state restoring for MQTT temperature sensors

* Adjust test

* Adjust test

cdce8p committed:
Add `workflow_dispatch` ci trigger (#66697)

emontnemery committed:
Improve MQTT device removal (#66766)

* Improve MQTT device removal

* Update homeassistant/components/mqtt/mixins.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Adjust tests

* Improve test coverage

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

MartinHjelmare committed:
Clean webostv notify (#66803)

* Replace conf with attr

* Test notify service without data parameter

* Clean kwargs access

* Replace icon constant

* Use data from notify

SaSa1983 committed:
Correct current temperature for tuya thermostats (#66715)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

SaSa1983 committed:
Correct current temperature for tuya thermostats (#66715)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

emontnemery committed:
Improve cleanup of Google Cast entities (#66801)

emontnemery committed:
Add support for removing Google Cast devices (#66808)

emontnemery committed:
Small cleanup of MQTT mixins (#66812)

cdce8p committed:
Add type ignore error codes [last ones] (#66816)

jeeftor committed:
Intellifire Diagnostic Sensors (#66597)

jkeljo committed:
Silence sisyphus chatty logging from dependencies (#66711)

eifinger committed:
Add google_travel_time sensor tests (#66568)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

kbickar committed:
Modernize Sleepiq and add new entities (#66336)

Co-authored-by: J. Nick Koston <nick@koston.org>

jeeftor committed:
Dependency Bump on Intellifire Lib (#66814)

balloob committed:
Bump aiohue to 4.2.1 (#66823)

balloob committed:
Bump aiohue to 4.2.1 (#66823)

balloob committed:
Bump frontend to 20220203.1 (#66827)

balloob committed:
Bumped version to 2022.2.9

balloob committed:
Bump hass-nabucasa to 0.53.0 (#66826)

madmat777 committed:
Add broadlink lb2 support (#63530)

* Update const.py

* Update to support LB2 version smart bulbs in Broadlink integration

* Update const.py

Added Space.

* Update updater.py

Updated so that LB2 lights use the LB1 update manager.

frenck committed:
Downgrade log warning->info for unregistered webhook message (#66830)

balloob committed:
Merge pull request #66829 from home-assistant/rc

gjohansson-ST committed:
Improve code quality workday (#66446)

* Code quality workday

* Modify from review

* Modify from review 2

* Fix mypy

jjlawren committed:
Initialize AlarmDecoder binary_sensor state as off instead of unknown (#65926)

epenet committed:
Add installed apps to samsungtv sources (#66752)

Co-authored-by: epenet <epenet@users.noreply.github.com>

jjlawren committed:
Bump SoCo to 0.26.3 (#66834)

bvweerd committed:
Fix eq3btsmart setting HVAC modes (#66394)

* Partly reverse preset incompatibility

It seems like some presets are unsupported by the native climate control of Home Assistant core. This change reverts the previous preset changes causing issues. It worked perfect with simple-thermostat custom lovelace card.

* Remove priority of preset above HVAC mode 

If a preset was available of the given command, the hvac mode change was ignored. This can result in HVAC settings are ignored. By removing the check for a preset, the preset does not supersede the HVAC mode anymore

* Revert "Partly reverse preset incompatibility"

This reverts commit 10fdc8eef457c369a042c631376ed33f29d0d8bf.

emontnemery committed:
Add support for validating and serializing selectors (#66565)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

cdce8p committed:
Add type ignore error codes [auth] (#66774)

kbickar committed:
Add SleepIQ device type (#66833)

Co-authored-by: J. Nick Koston <nick@koston.org>

balloob committed:
Bump hass-nabucasa to 0.53.1 (#66845)

bdraco committed:
Startup with an emergency self signed cert if the ssl certificate cannot be loaded (#66707)

balloob committed:
Validate in split_entity_id (#66835)

GitHub-Action committed:
[ci skip] Translation update

raman325 committed:
Create zwave_js ping button at the right time (#66848)

* Create zwave_js ping button at the right time

* fix tests

DurgNomis-drol committed:
Code enhancements for ISS (#66813)

* Code enhancements for ISS

* Assert options

ludeeus committed:
Add missing hass argument in async_request_config call (#66864)

cdce8p committed:
Don't run pytest CI jobs on push to forks (#66870)

gregsheremeta committed:
typo fix networrk --> network (#66878)

AnilDaoud committed:
Kaiterra type issue (#66867)

cdce8p committed:
Fix braviatv typing (#66871)

cdce8p committed:
Fix modbus typing (#66872)

bdraco committed:
Ensure dhcp can still discover new devices from device trackers (#66822)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

cdce8p committed:
Fix typo [recorder] (#66879)

janiversen committed:
Add support for INT8 and UINT8 in Modbus (#66889)

cdce8p committed:
Use assignment expressions [K-Z] (#66881)

cdce8p committed:
Use assignment expressions [A-I] (#66880)

cdce8p committed:
Use assignment expressions [other] (#66882)

bdraco committed:
Fix yeelight config flow ip update and timeout (#66883)

thecode committed:
Add Shelly gen2 error sensors (#66825)

klaasnicolaas committed:
Add Pure Energie integration (#66846)

trekie86 committed:
Update srpenergy dependency to 1.3.6 (#66821)

kbickar committed:
Create button entities for SleepIQ (#66849)

gjohansson-ST committed:
Bump pysensibo to 1.0.4 (#66886)

jingsno committed:
Fix Mill Gen1 Climate Control (#66899)

Fixes Mill Gen1 Climate Control, so it correctly returns the current status of the heating element.

rytilahti committed:
Enable consumable sensors per default for xiaomi_miio vacuums (#66843)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

jjlawren committed:
Proxy Plex thumbnail images in media browser (#66702)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bdraco committed:
Bump aiodiscover to 1.4.8 (#66892)

jt-nti committed:
Update RSS feed template (#62966)

rklomp committed:
Prefix sma sensor name (#65234)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

gjohansson-ST committed:
Bump pysensibo to v1.0.5 (#66906)

austinmroczek committed:
bump total_connect_client to 2022.2.1 (#66907)

ctalkington committed:
Update rokuecp to 0.14.1 (#66894)

GitHub-Action committed:
[ci skip] Translation update

bdraco committed:
Add support for WiZ diagnostics (#66817)

allenporter committed:
Update nest climate set_temperature to allow hvac_mode (#66909)

janiversen committed:
Remove async_setup_component() from tests (#66905)

chishm committed:
Add media browser support to dlna_dmr (#66425)

chemelli74 committed:
Fritz: fix unbound topology (#66877)

Co-authored-by: Michael <35783820+mib1185@users.noreply.github.com>
Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

emontnemery committed:
Deduplicate code in cast media_player (#66815)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

raman325 committed:
Improve zwave_js notification event handling (#66790)

balloob committed:
Bump frontend to 20220220.0 (#66919)

ludeeus committed:
Bump aiogithubapi from 22.2.0 to 22.2.3 (#66924)

oischinger committed:
Add service configuration URL for vicare (#66927)

frenck committed:
Introduce const file in LaMetric (#66929)

raman325 committed:
Add zwave_js.event automation trigger (#62828)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

janiversen committed:
Set slave default to 0, as already documented in Modbus (#66921)

ludeeus committed:
Use GraphQL for GitHub integration (#66928)

ludeeus committed:
Enable all GitHub sensors by default (#66931)

gjohansson-ST committed:
Bump pysensibo to v1.0.6 (#66930)

ludeeus committed:
Add Latest Tag sensor to GitHub integration (#66932)

timmo001 committed:
Add discussions sensors to GitHub (#66937)

chishm committed:
Avoid accessing hass.data in test_play_media_didl_metadata (#66939)

thecode committed:
Bump aiowebostv to 0.1.3 (#66942)

dewdropawoo committed:
Fix broken aftership sensor after pyaftership 21.11.0 bump (#66855)

* Fix update to pyaftership 21.11.0

Variable name collision and a missing property access was causing this sensor to always return zero elements.

* Move subscripting out of awaited statement

klaasnicolaas committed:
Update Pure Energie integration (#66946)

* Remove service entry_type

* Set raise on progress

rikroe committed:
Bump bimmer_connected to 0.8.11 (#66951)

Co-authored-by: rikroe <rikroe@users.noreply.github.com>

CoMPaTech committed:
Plugwise bump module version to fix heating-state and OnOff devices (#66936)

GitHub-Action committed:
[ci skip] Translation update

davet2001 committed:
Use new enums in vizio tests (#62710)

* Use new enums in vizio tests

* Code review: revert wrong conftest changes

* Code review: Revert incorrect removal of vizio const

* Reinstate wrongly reverted files

* Fix double line

* Fix new test after rebase

epenet committed:
Use hass.add_job in samsungtv (#66976)

Co-authored-by: epenet <epenet@users.noreply.github.com>

julesxxl committed:
Fix typo in const.py (#66856)

emontnemery committed:
Add WS API for removing a config entry from a device (#66188)

* Add WS API for removing a config entry from a device

* Address review comments

* Address review comments

* Remove entity cleanup from ConfigEntries

* Update + add tests

* Improve comments in test

* Add negative test

* Refactor according to review comments

* Add back async_remove_config_entry_device

* Remove unnecessary error handling

* Simplify error handling

epenet committed:
Bump samsungtvws to 1.7.0 (#66978)

Co-authored-by: epenet <epenet@users.noreply.github.com>

G-Two committed:
Bump to subarulink 0.4.2 (#66403)

frenck committed:
Merge branch 'master' into dev

G-Two committed:
Merge branch 'fix_sensors' of github.com:G-Two/home-assistant into fix_sensors

homeassistant commented:
Hi @GitHub-Action, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#65354).

Once you do that we will be able to review and accept this pull request.

Thanks!

G-Two commented:
Sorry, I messed up a rebase/commit.
